src/megaapi_impl.cpp

```diff
@@ -13275,28 +13278,18 @@ void MegaApiImpl::fa_complete(handle, fatype, const char* data, uint32_t len)
 
         tag = int(request->getNumber());
 
-        FileAccess *f = client->fsaccess->newfileaccess();
+        auto f = client->fsaccess->newfileaccess();
         string filePath(request->getFile());
         string localPath;
         fsAccess->path2local(&filePath, &localPath);
-
         fsAccess->unlinklocal(&localPath);
-        if(!f->fopen(&localPath, false, true))
-        {
-            delete f;
-            fireOnRequestFinish(request, MegaError(API_EWRITE));
-            continue;
-        }
 
-        if(!f->fwrite((const byte*)data, len, 0))
-        {
-            delete f;
-            fireOnRequestFinish(request, MegaError(API_EWRITE));
-            continue;
-        }
+        bool success = f->fopen(&localPath, false, true)
+                    && f->fwrite((const byte*)data, len, 0);
 
-        delete f;
-        fireOnRequestFinish(request, MegaError(API_OK));
+        f.reset();
```

Comment:
Yes, I thought it's better to close the file handle before calling `fireOnRequestFinish`, since the 'fire' functions call the application back, and the client may want to open the file with its own handle etc.